spec/unit/provider/debian_spec.rb

```diff
@@ -0,0 +1,136 @@
+#
+# Author:: Kapil Chouhan (<kapil.chouhan@msystechnologies.com>)
+# Copyright:: Copyright 2008-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+describe Chef::Provider::Package::Debian do
+  let(:node) do
+    node = Chef::Node.new
+    node.automatic_attrs[:platform] = :just_testing
+    node.automatic_attrs[:platform_version] = :just_testing
+    node
+  end
+  let(:events) { Chef::EventDispatch::Dispatcher.new }
+  let(:logger) { double("Mixlib::Log::Child").as_null_object }
+  let(:run_context) { Chef::RunContext.new(node, {}, events) }
+  let(:new_resource) { Chef::Resource::AptPackage.new("emacs", run_context) }
+  let(:current_resource) { Chef::Resource::AptPackage.new("emacs", run_context) }
```

Comment:
Can the abstract `MyDebianPackageResource` and `MyDebianPackageProvider` classes that appear later in this spec file be used here instead of depending on `AptPackage`?

Reply:
@robbkidd this test case is related to action `reconfig` and this action is used by only `apt_package` and `dpkg_package`, so currently we are checking logger like this `apt_package[emacs] is NOT installed - nothing to do` if we use the abstract `MyDebianPackageResource` and `MyDebianPackageProvider` classes then we got logger like this `[emacs] is NOT installed - nothing to do`

Reply:
@robbkidd In specs we are setting `response_file` property, and this property is only present in `apt` and `dpkg`, so that's why here I am using `AptPackage`